src/array/array.cc

```diff
@@ -350,6 +350,16 @@ void CSRSort_(CSRMatrix* csr) {
   });
 }
 
+IdArray CSRSortByTag_(CSRMatrix* csr, IdArray tag, int64_t num_tags) {
+  IdArray ret;
```

Comment:
Please add some sanity checks. E.g., the length of the tag array should be equal to the number of columns; csr and tag should have the same context; tag array must be integer etc. You can see other operators for reference.